README.md

```diff
@@ -1,9 +1,66 @@
 
+# Accessing the re-gridded files
+## On Caldera
+On Caldera, re-gridded file paths are structured like this:`/caldera/projects/usgs/water/impd/pump/gridmet/drb_gridmet_tools/drb-gridmet/{fabric}/{run_date}/drb_climate_{run_date}.nc`. For example: ``/caldera/projects/usgs/water/impd/pump/gridmet/drb_gridmet_tools/drb-gridmet/nhm/2022_06_14/drb_climate_2022_06_14.nc`.
 
-## Selected gridMET variables: 
+## On S3
+As of 6/2022, the results of the workflow are, by default, also stored S3 in the `drb-gridmet` bucket. For example, `s3://drb-gridmet/nhm/2022_06_14/drb_climate_2022_06_14.nc`.:w
+
+# Running re-gridding for the Delaware River Basin via Snakemake 
+The re-gridding process for the Delaware River Basin has been run on USGS's Tallgrass via Singularity. It has been run for the National Hydrologic Model (NHM) fabric and the National Hydrographic Database (NHD) fabric. It should be able to be run via Docker as well as Singularity, but it has not been tried yet. Instructions for running and modifying the pipeline are below.
+
+_Parallelization in Snakemake_
+The Snakemake workflow parallelizes the re-gridding of the 8 Gridmet variables. As long as you provide at least 8 cores, these tasks will run in parallel.
+
+_S3_
+By default, the results of the workflow are stored locally and on S3 in the `drb-gridmet` bucket. If you would like to only store the resulting files locally, change the `use_S3` option in the config file to `False`.
+
+NOTE: Although this workflow was first run on DRB catchments, it should be able to be run on any set of polygons in the conterminous US. This however, has not be tested.
+
+## Run via Singularity
+To run the workflow with Singularity, you can either use the image already in Caldera via Tallgrass, or pull the image into a new directory.
+
+### Option A. Use the existing image and cloned repo
+1. move to the correct directory
+
+```
+cd /caldera/projects/usgs/water/impd/pump/gridmet/drb_gridmet_tools/
+```
+
+2. Decide on which fabric you want to use. You will use a different `config` file depending on which fabric you use: `config_nhm.yml` for the NHM fabric and `config_nhd.yml` for the `nhd` fabric.
+3. [Optional] edit the options in the config file 
+4. Run the workflow
+You can run the workflow on Tallgrass either in batch mode or interactively. Subtite your HPC account (may not be 'iidd') and desired config file (may not be `config_nhd.yml`)
+
+You can run the workflow via `sbatch` 
+```
+sbatch -A iidd slurm/launch_snakemake.slum config_nhd.yml
+```
+It may be helpful instead to run it interactively
+
+```
+salloc -N 1 -n 8 -t 10:00:00 -p cpu -A iidd
+module load singularity
+singularity exec-agg_v0.3.sif /opt/conda/bin/snakemake -j --configfile config_nhd.yml 
```

Comment:
There isn't write access to all the files / directories (see below) so everyone but Jeff will get this error. We'll have to ask HPC folks to give group permissions in this folder because I'm pretty sure Jeff doesn't have access to USGS HPC anymore 

```
[jzwart@tg-login1 drb_gridmet_tools] ls -la
total 995356
drwxr-sr-x 11 jsadler pump       4096 Jun 17 12:24 .
drwxr-sr-x  3 jsadler pump       4096 Jan 28  2022 ..
drwxr-sr-x  3 jsadler pump       4096 Jun 17 12:22 archive
-rw-r--r--  1 jsadler pump        509 Jun 15 13:43 config_nhd.yml
-rw-r--r--  1 jsadler pump        536 Jun 15 16:04 config_nhm.yml
drwxr-sr-x  4 jsadler pump       4096 Jun 14 11:09 drb-gridmet
-rw-r--r--  1 jsadler pump        536 Apr  8 11:42 environment.yml
drwxr-sr-x  8 jsadler pump       4096 Jun 17 12:24 .git
-rw-r--r--  1 jsadler pump        123 Jan 28  2022 .gitignore
-rw-r--r--  1 jsadler pump       3079 Jun 15 12:34 gridmet_aggregation_PRMS.py
-rwxrwxr-x  1 jsadler pump 1019154432 Jun 15 13:40 gridmet-agg_v0.3.sif
-rw-r--r--  1 jsadler pump       7329 Jun 14 15:51 gridmet_split_script.py
drwxr-sr-x  2 jsadler pump       4096 Apr 21 15:57 .ipynb_checkpoints
drwxr-sr-x  2 jsadler pump       4096 Jun 15 12:34 __pycache__
-rw-r--r--  1 jsadler pump       4081 Jun 15 16:06 README.md
drwxr-sr-x  5 jsadler pump       4096 Jun 15 13:46 scratch
drwxr-sr-x  2 jsadler pump       4096 Jun 17 12:23 slurm
drwxr-sr-x  2 jsadler pump       4096 Jun 17 12:23 slurm_out
-rw-r--r--  1 jsadler pump       6400 Jun 15 16:02 Snakefile
drwxr-sr-x 11 jsadler pump       4096 Jun  9 16:57 .snakemake
```

Reply:
I'll ask in #tallgrass-help channel

Reply:
Yes (see [here](https://teams.microsoft.com/l/message/19:27fb59666e1b47b5ba2cb658c6370f24@thread.skype/1663843627639?tenantId=0693b5ba-4b18-4d7b-9341-f32f400a5494&groupId=0d997551-abe8-4b31-aa0a-06bb8ad40463&parentMessageId=1663843627639&teamName=GS-HPCUsers&channelName=Tallgrass%20Help&createdTime=1663843627639&allowXTenantAccess=false)), they changed who owns the files and permissions should be updated. 

```
[jzwart@tg-login2 drb_gridmet_tools] ls -la
total 995356
drwxrwsr-x 11 jzwart pump       4096 Jun 17 12:24 .
drwxrwsr-x  3 jzwart pump       4096 Jan 28  2022 ..
drwxrwsr-x  3 jzwart pump       4096 Jun 17 12:22 archive
-rw-rwxr--  1 jzwart pump        509 Jun 15 13:43 config_nhd.yml
-rw-rwxr--  1 jzwart pump        536 Jun 15 16:04 config_nhm.yml
drwxrwsr-x  4 jzwart pump       4096 Jun 14 11:09 drb-gridmet
-rw-rwxr--  1 jzwart pump        536 Apr  8 11:42 environment.yml
drwxrwsr-x  8 jzwart pump       4096 Jun 17 12:24 .git
-rw-rwxr--  1 jzwart pump        123 Jan 28  2022 .gitignore
-rw-rwxr--  1 jzwart pump       3079 Jun 15 12:34 gridmet_aggregation_PRMS.py
-rwxrwxr-x  1 jzwart pump 1019154432 Jun 15 13:40 gridmet-agg_v0.3.sif
-rw-rwxr--  1 jzwart pump       7329 Jun 14 15:51 gridmet_split_script.py
drwxrwsr-x  2 jzwart pump       4096 Apr 21 15:57 .ipynb_checkpoints
drwxrwsr-x  2 jzwart pump       4096 Jun 15 12:34 __pycache__
-rw-rwxr--  1 jzwart pump       4081 Jun 15 16:06 README.md
drwxrwsr-x  5 jzwart pump       4096 Jun 15 13:46 scratch
drwxrwsr-x  2 jzwart pump       4096 Jun 17 12:23 slurm
drwxrwsr-x  2 jzwart pump       4096 Jun 17 12:23 slurm_out
-rw-rwxr--  1 jzwart pump       6400 Jun 15 16:02 Snakefile
drwxrwsr-x 11 jzwart pump       4096 Jun  9 16:57 .snakemake
```